--- ./src/sudo.c.orig	2025-11-04 15:29:57.987524369 +0100
+++ ./src/sudo.c	2025-11-04 15:30:10.472518192 +0100
@@ -620,7 +620,7 @@ get_user_info(struct user_details *ud)
 
     ttydev = get_process_ttyname(path, sizeof(path));
     if (ttydev != NODEV) {
-	if (asprintf(&info[++i], "ttydev=%lld", (long long)ttydev) == -1)
+	if (asprintf(&info[++i], "ttydev=%llu", (unsigned long long)ttydev) == -1)
 	    goto oom;
 	/* The terminal device file may be missing in a chroot() jail. */
 	if (path[0] != '\0') {
